ci/travis/doxygen.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+echo_red() { printf "\033[1;31m$*\033[m\n"; }
+echo_green() { printf "\033[1;32m$*\033[m\n"; }
+
+############################################################################
+# Check if the documentation will be generated w/o warnings or errors
+############################################################################
+pushd ${TRAVIS_BUILD_DIR}/doc
+(cd build && ! make doc 2>&1 | grep -E "warning|error") || {
+        echo_red "Documentation incomplete or errors in the generation of it have occured!"
+        exit 1
+}
+popd
+
+echo_green "Documentation was generated successfully!"
+
+############################################################################
+# If the current build is not a pull request and it is on master the 
+# documentation will be pushed to the gh-pages branch
+############################################################################
+if [[ "${TRAVIS_PULL_REQUEST}" == "false" && "${TRAVIS_BRANCH}" == "master" ]]
+then
+        pushd ${TRAVIS_BUILD_DIR}/doc
+        git clone https://amiclaus:${GITHUB_TOKEN}@github.com/analogdevicesinc/no-OS --depth 1 --branch=gh-pages doc/html &>/dev/null
```

Comment:
I still don't think that's a good idea to use personal GitHub accounts for these tasks.

Reply:
I added an encrypted access token for these tasks.